swift/ql/lib/codeql/swift/frameworks/StandardLibrary/String.qll

```diff
@@ -125,6 +130,15 @@ private class StringSummaries extends SummaryModelCsv {
         ";String;true;encode(to:);;;Argument[-1];Argument[0];taint",
         ";String;true;decodeCString(_:as:repairingInvalidCodeUnits:);;;Argument[0];ReturnValue.TupleElement[0];taint",
         ";String;true;decodeCString(_:as:repairingInvalidCodeUnits:);;;Argument[0].CollectionElement;ReturnValue.TupleElement[0];taint",
+        ";String;true;withUTF8(_:);;;Argument[-1];Argument[0].Parameter[0].CollectionElement;taint",
+        ";String;true;withUTF8(_:);;;Argument[0].Parameter[0].CollectionElement;Argument[-1];taint",
+        ";String;true;withUTF8(_:);;;Argument[0].ReturnValue;ReturnValue;value",
+        ";String;true;withPlatformString(_:);;;Argument[-1];Argument[0].Parameter[0].CollectionElement;taint",
+        ";String;true;withPlatformString(_:);;;Argument[0].ReturnValue;ReturnValue;value",
```

Comment:
Same question here.

Reply:
I don't think you can mutate the string with these versions.  I'll check...

Reply:
Sorry for taking a while to get back to this.  I think it's only possible to modify the object using the methods marked `mutating`, that is `withMutableCharacters` and `withUTF8` in this PR (and I'm not sure it's practically possible with the latter, see the [doc](https://developer.apple.com/documentation/swift/string/withutf8(_:))).

Some experimentation [here](https://swiftfiddle.com/?c=H4sIAAAAAAAAA82VUWvCMBDHv8qRp1akVedgK3MPEwQfBoLzadlDpqmG2bQkV3WI332JabfqpkzGoA%2Bl4Z%2FL%2Fe%2Fu19AtmZKIiCRLFcIgzeWMoUgllYU0ftfIkxGbvrE5p0ZfMQUaIhijEnIOPaCEMUaJ3cuMhB4l2srU0z4lvtV1sBa46Lsj3pZKgAwVCGlXS46Q5Mhel3xkxB5MpGYxfyyk1KTk6q7fXzB179lANEkim8DfJyo89ymfWy%2FO2q2dfzXmy%2Bgz9EAqK65GBnlmhsI9xTNemLc7rSZMzbQwgqsjCwAXD16l6bKSY%2F%2B2M%2FuPJnb%2BOSRhCF6uOYzXIkboBp2g65egRkuGcaqSP%2FEa2leaBWWymvFr%2F5Lf4Sxqh9EBmzwNbi7D9JDHMVclrImBdVMjOJ2zSGy3tQPxw30C890ALoSGVHITFDYcrQKBvRBsasavj9Gdrt48gUamcChnfFMZqN0QUnNlz20oaQLDPcZK%2BNmhfiuqkvuyck7PqRFSSZpkZX4418Et2X0AY3tHiHwGAAA%3D) - notice that the changes are never reflected back in `s`.